sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -208,6 +208,7 @@ static int _fill_ipv6_hdr(kernel_pid_t iface, ng_pktsnip_t *ipv6,
 
     DEBUG("ipv6: calculate checksum for upper header.\n");
 
+#if NG_NETIF_NUMOF > 1
     if (payload->users > 1) {
```

Comment:
We only need to duplicate the payload for checksum calculation (what happens here) if there are multiple interfaces (since this will result in different source addresses in the header).